pumps/hybrid.go

```diff
@@ -39,18 +35,71 @@ var (
 			return nil
 		},
 	}
+	GlobalRPCCallTimeout = 30
+	ErrRPCLogin          = errors.New("RPC login incorrect")
 )
 
 // HybridPump allows to send analytics to MDCB over RPC
 type HybridPump struct {
-	aggregated                 bool
-	trackAllPaths              bool
-	storeAnalyticsPerMinute    bool
-	aggregationTime            int
-	enableAggregateSelfHealing bool
-	ignoreTagPrefixList        []string
 	CommonPumpConfig
-	rpcConfig rpc.Config
+
+	clientSingleton   *gorpc.Client
+	dispatcher        *gorpc.Dispatcher
+	clientIsConnected atomic.Value
+
+	funcClientSingleton *gorpc.DispatcherClient
+
+	hybridConfig *HybridPumpConf
+}
+
+// @PumpConf Hybrid
+type HybridPumpConf struct {
+	EnvPrefix string `mapstructure:"meta_env_prefix"`
+
+	// RPC server connection string
+	ConnectionString string `mapstructure:"connection_string"`
+	// RPC server key
+	RPCKey string `mapstructure:"rpc_key"`
+	// RPC server API key
+	APIKey string `mapstructure:"api_key"`
+
+	// Specifies prefixes of tags that should be ignored if `aggregated` is set to `true`.
+	IgnoreTagPrefixList []string `json:"ignore_tag_prefix_list" mapstructure:"ignore_tag_prefix_list"`
+
+	// RPC server call timeout
+	CallTimeout int `mapstructure:"call_timeout"`
+	// RPC server connection pool size
+	RPCPoolSize int `mapstructure:"rpc_pool_size"`
+
+	aggregationTime int
```

Comment:
can we add a comment on this field? (despite it's unexported)

Reply:
Suggested to make it a func in another place; the value hides logic.

Reply:
added comment